app/lib/admin/metrics/measure/query_helper.rb

```diff
@@ -15,19 +15,49 @@ def sanitized_sql_string
     ActiveRecord::Base.sanitize_sql_array(sql_array)
   end
 
-  def generated_series_days
-    Arel.sql(
-      <<~SQL.squish
+  def sql_array
+    [sql_query_string, { start_at: @start_at, end_at: @end_at }]
+  end
+
+  def sql_query_string
+    <<~SQL.squish
+      SELECT axis.*, (
+        WITH data_source AS (#{data_source.to_sql})
```

Comment:
`to_sql` runs the risk of messing with binds, and I wish we could avoid the back-and-forth between raw SQL and ActiveRecord relations. This kind of makes me think we could use `ActiveRecord::QueryMethods#with`, but it's not obvious how.

Reply:
I started on that last thing (using `with` here) for that exact reason ... the fact that we have this constructed sql string/array approach in the outer layer here, and then all the various report classes are all just return `Relation` objects is the final piece of this refactor, but a) I also couldnt find an obvious way to do that, b) this was already pretty large.

I still think this is a net-improvement vs previous, but I would definitely pursue that aspect as followup.